tests/python_functional/conftest.py

```diff
@@ -63,18 +80,27 @@ def pytest_collection_modifyitems(config, items):
             item.add_marker(skip_slow)
 
 
-def runwithvalgrind(request):
-    return request.config.getoption("--run-with-valgrind")
-
+def pytest_runtest_logreport(report):
+    if report.outcome == "failed":
+        logger.error("\n{}".format(report.longrepr))
 
-def get_relative_report_dir():
-    return str(Path("reports/", get_current_date()))
 
+@pytest.hookimpl(hookwrapper=True, tryfirst=True)
+def pytest_runtest_setup(item):
+    def construct_report_file_path(item):
+        relative_report_dir = item._request.config.getoption("--reports")
+        testcase_name = item._request.node.name
+        file_name = "testcase_{}.log".format(testcase_name)
+        return Path(relative_report_dir, testcase_name, file_name).absolute()
 
-def get_current_date():
-    return datetime.now().strftime("%Y-%m-%d-%H-%M-%S-%f")
+    config = item.config
+    logging_plugin = config.pluginmanager.get_plugin("logging-plugin")
+    report_file_path = construct_report_file_path(item)
+    logging_plugin.set_log_path(report_file_path)
+    yield
```

Comment:
Is this `yield` needed?

Reply:
Yes, it is needed to returning from the pytest hook (this is a pytest internal thing)